srcs/validator.cc

```diff
@@ -0,0 +1,40 @@
+#include <queue>
+#include <vector>
+#include <fruit/fruit.h>
+#include "exception.hpp"
+#include "token.hpp"
+#include "validator.hpp"
+
+class ParaValidator: public Validator{
+    public:
+    using Inject = ParaValidator();
+    ParaValidator() = default;
+    virtual Result Validate (std::queue<PToken> tokenList) override;
+    virtual ~ParaValidator()=default;
+    private:
+        std::vector<std::size_t> _posstack;
+};
+
+Result ParaValidator::Validate(std::queue<PToken> tokenList){
```

Comment:
If you take the entire token list as input, the class isn't stateful and the stack will be local to the method.

Reply:
Can discuss this offline a bit more on whether validator is unnecessary. As for the purpose of this PR, I will keep the validator as is.